middle_end/closure/closure_middle_end.ml

```diff
@@ -35,14 +35,23 @@ let lambda_to_clambda ~backend ~filename:_ ~prefixname:_ ~ppf_dump
     Closure.intro ~backend ~size:lambda.main_module_block_size lambda.code
   in
   let provenance : Clambda.usymbol_provenance =
+    let current_unit_ident =
+      Compilation_unit.get_current_exn ()
+      |> Compilation_unit.name
+      |> Compilation_unit.Name.to_string
+      |> Ident.create_persistent
+    in
     { original_idents = [];
-      module_path =
-        Path.Pident (Ident.create_persistent (Compilenv.current_unit_name ()));
+      module_path = Path.Pident current_unit_ident;
```

Comment:
See review comment in Compilenv about this `module_path`.

Reply:
Added another CR.